shell/src/main/java/alluxio/cli/docgen/ConfigurationDocValidator.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.cli.docgen;
+
+import static alluxio.cli.docgen.ConfigurationDocGenerator.CSV_FILE_NAMES;
+import static alluxio.cli.docgen.ConfigurationDocGenerator.YML_FILE_NAMES;
+
+import alluxio.annotation.PublicApi;
+import alluxio.conf.Configuration;
+import alluxio.conf.PropertyKey;
+import alluxio.util.io.PathUtils;
+
+import org.apache.commons.lang3.StringEscapeUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Pattern;
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * Utility to validate existence of any necessary configuration docGen changes during pull request.
+ */
+@ThreadSafe
+@PublicApi
+public final class ConfigurationDocValidator extends DocValidator {
+  private static final Logger LOG = LoggerFactory.getLogger(ConfigurationDocValidator.class);
+  public static final String CSV_FILE_HEADER = "propertyName,defaultValue";
+
+  private ConfigurationDocValidator() {
+  } // prevent instantiation
+
+  /**
+   * Validates configuration files.
+   *
+   * @return returns number of deltas
+   */
+  public static Integer validate() throws IOException {
+    Collection<? extends PropertyKey> defaultKeys = PropertyKey.defaultKeys();
+    defaultKeys.removeIf(PropertyKey::isHidden);
+
+    // Sort defaultKeys
+    List<PropertyKey> dfkeys = new ArrayList<>(defaultKeys);
+    Collections.sort(dfkeys);
+
+    Integer numDeltas = 0;
+    numDeltas += validateFiles(dfkeys, CSV_FILE_DIR, CSV_FILE_NAMES);
+    numDeltas += validateFiles(dfkeys, YML_FILE_DIR, YML_FILE_NAMES);
+
+    return numDeltas;
+  }
+
+  private static Integer validateFiles(List<PropertyKey> dfkeys,
+                                       String fileDir, List<String> fileNames)
+          throws IOException {
+
+    String filePath = PathUtils.concatPath(Configuration.getString(PropertyKey.HOME), fileDir);
+    Integer numDeltas = 0;
+    for (String fileName : fileNames) {
+      String key = fileName.substring(0, fileName.indexOf("configuration") - 1);
+
+      // based on key substring within each filename, create a regex to
+      // filter for specific property keys.
+      List<String> rPatterns = new ArrayList<String>();
```

Comment:
why are regex patterns used here whereas the generation logic is using string prefixes? shouldn't they be the same?